jenkins/JenkinsFile-PyPI-packaging

```diff
@@ -0,0 +1,278 @@
+        TWINE_NON_INTERACTIVE = 1
+        TWINE_REPOSITORY = "pypi"
+        TWINE_USERNAME = "__token__"
+        TWINE_PASSWORD = credentials("pypi-api-key")
+    }
+
+    options {
+        timeout(time: 1, unit: "HOURS")
+    }
+
+    parameters {
+        string(
+            name:"TLCPACK_GIT_REPO_URL",
+            defaultValue: "https://github.com/tlc-pack/tlcpack",
+            description: "URL for the tlcpack repository")
+        string(
+            name:"TLCPACK_CPU_TAG",
+            defaultValue: "2fddbe0",
+            description: "Tag for the Docker tlcpack CPU image")
+        string(
+            name:"TLCPACK_AARCH64_TAG",
+            defaultValue: "3e5b139d",
+            description: "Tag for the Docker tlcpack AArch64 image")
+        string(
+            name:"TVM_GIT_REV",
+            defaultValue: "main",
+            description: "Git revision to checkout.")
+        string(
+            name:"TVM_GIT_REPO_URL",
+            defaultValue: "https://github.com/apache/tvm",
+            description: "URL for the TVM repository")
+    }
+
+    stages {
+
+        stage("Setup workspace") {
+            agent {
+                node { label "CPU" }
+            }
+            steps {
+                cleanWs()
+                clone_tlcpack()
+                clone_tvm()
+                dir("tvm") {
+                    // Checkout the requested reference, so that we
+                    // can build images from branches/tags
+                    sh (
+                        script: "git checkout ${params.TVM_GIT_REV}",
+                        label: "Checkout requested reference",
+                    )
+                    script {
+                        TVM_CURRENT_SHORT_REF = sh (
+                            script: "git rev-parse --short HEAD",
+                            label: "Set current build display name",
+                            returnStdout: true
+                            ).trim()
+                        currentBuild.displayName = "tlcpack-${TVM_CURRENT_SHORT_REF}"
+                    } // script
+                } // dir(tvm)
+                sync_package()
+                stash (
+                    includes: "*/**",
+                    name: "tvm_workspace",
+                    allowEmpty: false,
+                )
+            }
+        } // stage(Setup workspace)
+
+        stage("Build and test wheels") {
+            parallel {
+
+                stage("CPU wheel") {
+                    agent {
+                        node { label "CPU" }
+                    }
+                    steps {
+                        cleanWs();
+                        unstash "tvm_workspace"
+                        script {
+                            pull_docker_image("${CPU_IMAGE}")
+                            build_wheels("${CPU_IMAGE}")
+                            cleanup_docker_image("${CPU_IMAGE}")
+                        } // script
+                        stash (
```

Comment:
I'm not a huge fan of `stash` / `unstash` since it applies a lot of pressure to the jenkins head node (where the stashes have to be uploaded / downloaded from) and the head node is also short on disk space so stuffing it up with images would eat that up pretty fast. The regular TVM CI uses S3 for artifacts, can we use that here too?

Reply:
+1 to this

Reply:
Thanks for the feedback, I will look in to using S3 buckets for the artefacts. We aren't using S3 in our local environment, but I can try to make a modification to set it as the default option for this script, instead of stashing.